modules/account-lib/src/coin/near/contractCallTxBuilder.ts

```diff
@@ -0,0 +1,41 @@
+import { TransactionBuilder } from './transactionBuilder';
+import BN from 'bn.js';
+import * as nearAPI from 'near-api-js';
+import { BuildTransactionError } from '../baseCoin/errors';
+
+export interface functionCallParams {
+  methodName: string;
+  args: { amount: string };
+  gas: BN;
+  deposit: BN;
+}
+
+export abstract class ContractCallTxBuider extends TransactionBuilder {
+  protected _methodName: string;
+  protected _args: { amount: string };
+  protected _gas: BN;
+  protected _deposit: BN;
+
+  functionCallParams(params: functionCallParams): this {
+    this._methodName = params.methodName;
+    this._args = params.args;
+    if (!this.isValidAmount(params.gas)) {
+      throw new BuildTransactionError(`Invalid gas amount ${params.gas}`);
+    }
+    this._gas = params.gas;
+    if (!this.isValidAmount(params.deposit)) {
+      throw new BuildTransactionError(`Invalid deposit amount ${params.deposit}`);
+    }
+    this._deposit = params.deposit;
+    return this;
+  }
```

Comment:
lets change it, for inherited builders we already know which is the method name so we have two alternatives, do this builder totally agnostic and use an object like the current implementation, or make methods in order to let inherited builders set those params.

e.g
StakingActivateBuilder().amount('100').gas('100')
StakingDeactivateBuilder().amount('100').gas('100')
StakingWithdrawBuilder().amount('100').gas('100')

Is code responsability check if we need internally set the .amount() on the _deposit or the _methodName calling the ContractCallBuilder.
Is staking builder responsability set the method name ( we already know it )